packages/network-of-terms-reconciliation/src/query.ts

```diff
@@ -36,19 +38,41 @@ export async function reconciliationQuery(
       );
       const terms = termsResult instanceof Terms ? termsResult.terms : [];
       results[queryId] = {
-        result: terms.map(term => ({
-          id: term.id.value.toString(),
-          name: term.prefLabels.map(label => label.value).join(' • '), // Join similarly to network-of-terms-demo.
-          score: 1, // Hard-coded score for now because scoring will have to be done on our side because no scores are returned from the sources.
-          description: term.altLabels.map(label => label.value).join(' • '),
-        })),
+        result: terms
+          .map(term => ({
+            id: term.id.value.toString(),
+            name: term.prefLabels.map(label => label.value).join(' • '), // Join similarly to network-of-terms-demo.
+            score: score(queryString, term),
+            description: term.altLabels.map(label => label.value).join(' • '),
+          }))
+          .sort((a, b) => b.score - a.score),
       };
       return results;
     },
     Promise.resolve({})
   );
 }
 
+/**
+ * Calculate score for term based on case-insensitive Levenshtein distance between the query string and the term’s
+ * prefLabels.
+ */
+const score = (queryString: string, term: Term): number => {
+  if (term.prefLabels.length === 0) {
+    return 0;
+  }
+
+  return (
+    1 /
+    term.prefLabels.reduce(
+      (distance, prefLabel) =>
+        distance +
+        leven(queryString.toLowerCase(), prefLabel.value.toLowerCase()),
+      1
+    )
+  );
+};
```

Comment:
Do these scores make sense?

- identical word = score 1
- any differences decrease that score to fractions of 1, e.g. 0.35

The Reconciliation spec doesn’t specify that the scores must look like, but their examples seem to be on a score of 1–100(?). Which scores are better readable for humans? See also the OpenRefine screenshot above.

Reply:
Good one!

I don't have a strong opinion about this, but I'd prefer 1-100: it makes it easier to say that word 'hout' matches label 'hout' 100% and label 'Maashout' 50%.

Reply:
Yep, let’s stick with percentages.